lib/timers.js

```diff
@@ -232,6 +231,12 @@ function listOnTimeout() {
   } else if (list === refedLists[msecs]) {
     delete refedLists[msecs];
```

Comment:
> What you're suggesting would indeed fix the problem, but it seems to me it would lead to a less robust solution that would be more difficult to understand.

I agree this.  
And it is the point that we don't have an agreement with @Fishrock123 from his comment of 
> There are more obscure parts of the timers code imo so I don't think it is an issue.